Setup and use HTTPS/TLS on the Management API

frozen3 opened the pull request:
This is my first attempt at getting the module working using https for management.  This could potentially resolve #207.  

- Works with WF 8.2.1+, EAP 6/7. 
- Currently this only works with standalone mode.  It likely could be tweaked for domain mode, however I don't have much experience using domain mode.  Perhaps someone in the community could help.
- Does add a dependency to puppetlabs/java_ks.

jairojunior commented:
That's very interesting. I would just change two things:

- Use boolean `secure` instead of protocol, cause you can't actually select the protocol, just if you will use a secure layer or not.
- Replace openssl exec with camptocamp/openssl.

frozen3 commented:
@jairojunior I've replaced the custom exec with camptocamp/openssl and converted over to using a secure boolean value over the protocol string.  Take a look and let me know

jairojunior commented:
Sorry, just merged it.
